src/SmallStrainPlasticity/IsotropicHardening.jl

```diff
@@ -0,0 +1,26 @@
+# Isotropic hardening
+abstract type AbstractIsoHard{T} end
```

Comment:
We dont have any naming standard really, but i think i like full names better, AbstractIsotropicHardening.
I also think that the type parameter T is unnecessary at this level?

Reply:
OK, I have no good idea about naming conventions. So should all names be spelled out, e.g. IsotropicHardeningVoce (below)?

Reply:
I think the guidelines from the Julia docs are really good: https://docs.julialang.org/en/v1/manual/style-guide/#Use-naming-conventions-consistent-with-Julia-base/
I agree with Elias that full names are better as long as they don't become very unreasonably long.

Reply:
And regarding the type parameter: I think we had a discussion recently that it is not really needed to have a material with material parameters of any other type but `Float64` (it's just a parameter set, you will never e.g. differentiate the material itself). But if there is a good reason for having this Parameter I'll be happy to put it up for discussion again!

Reply:
My reason was type-stability, but perhaps there is another way of ensuring this that I'm not seeing. 
The idea is that some functions will have different dispatch depending on, for example, the number of back-stresses. Additionally, to avoid run-time function lookup (and issues with type stability), I found it necessary to explicitly dispatch on the type of e.g. kinematic hardening rule... Would of course be nice if this could be solved while keeping the material struct simpler!